test/libsolidity/semanticTests/structs/recursive_structs.sol

```diff
@@ -15,5 +15,7 @@ contract C {
     }
 }
 
+// ====
+// compileViaYul: also
```

Comment:
It might be nice to have another version of this test case that actually checks that the recursive struct is properly zeroed out...
Something like:
```
contract C {
    struct S {
        uint16 v;
        S[] x;
    }
    uint8[77] padding;
    S s;
    constructor() {
         s.v = 21;
         s.x = new S[](3); /* or s.x.push(); s.x.push(); s.x.push(); */
         s.x[0].v = 101; s.x[1].v = 102; s.x[2].v = 103;
    }
    function f() public returns (uint256 a, uint256 b, uint256 c, uint256 d) {
       S storage sptr1 = s.x[0];
       S storage sptr2 = s.x[1];
       S storage sptr3 = s.x[2];
       uint256 slot1; uint256 slot2; uint256 slot3;
       assembly { slot1 := sptr1.slot slot2 := sptr2.slot slot3 := sptr3.slot }
       delete s;
       assembly { a := sload(s.ptr) b := sload(slot1) c := sload(slot2) d := sload(slot3) }
    }
}
```

Reply:
(while confirming locally that this indeed returns ``21, 101, 102, 103``, if the ``delete`` is removed :-))

Reply:
(And if we feel paranoid we could plug in a second layer even, but I think one layer should be fine for this)

Reply:
This test doesn't work for sol->yul as there are some features missing (if I use push there is missing zeroValue for structs, and other way it is missing copying arrays from memory to storage, should be activated after we merge #9738 one day). I will leave this here for now, and will not compile via yul.
I checked, it indeed returns 21, 101, 102, 103 without delete in old codegen :)

Reply:
Hm - maybe the missing zero-value for structs (and thus allowing pushing them) could even be done before #9738 as well? It's again not really related to #9738 itself, but rather a prerequisite only...

Reply:
I agree it should be done before #9738, but would not include it in this PR honestly :)
But will work on that after we finish this one, it shouldn't be too hard to do it.

Reply:
Sounds good!

Reply:
#9914 implements pushing to array of structs, so this test can be added here now. I will unresolve this conversation and re-add the test :) Will wait for @ekpyron to mark as resolved